lib/spack/spack/util/path.py

```diff
@@ -157,20 +160,23 @@ def substitute_config_variables(path):
 
     Spack allows paths in configs to have some placeholders, as follows:
 
-    - $env                 The active Spack environment.
-    - $spack               The Spack instance's prefix
-    - $tempdir             Default temporary directory returned by tempfile.gettempdir()
-    - $user                The current user's username
-    - $user_cache_path     The user cache directory (~/.spack, unless overridden)
-    - $architecture        The spack architecture triple for the current system
-    - $arch                The spack architecture triple for the current system
-    - $platform            The spack platform for the current system
-    - $os                  The OS of the current system
-    - $operating_system    The OS of the current system
-    - $target              The ISA target detected for the system
-    - $target_family       The family of the target detected for the system
-    - $date                The current date (YYYY-MM-DD)
-    - $spack_short_version The spack short version
+    - $env                  The active Spack environment.
+    - $spack                The Spack instance's prefix
+    - $tempdir              Default temporary directory returned by tempfile.gettempdir()
+    - $user                 The current user's username
+    - $user_cache_path      The user cache directory ($XDG_DATA_HOME, unless overridden)
```

Comment:
```suggestion
    - $user_cache_path      The user cache directory ($[SPACK,XDG]_DATA_HOME, unless overridden)
```
Not sure if this is clear, but is this the case or is this an exception? I think it should be the case.